test/gtest/ucs/test_datatype.cc

```diff
@@ -135,6 +135,9 @@ UCS_TEST_F(test_datatype, hlist_basic) {
     ucs_hlist_add_head(&head, &elem1.hlist);
     elem = ucs_list_extract_head_elem(&head, elem_t, hlist);
     EXPECT_EQ(&elem1, elem);
+    /* extract head element, it has to be NULL, since the hlist is empty */
+    elem = ucs_list_extract_head_elem(&head, elem_t, hlist);
```

Comment:
BTW can u pls rename `ucs_list_extract_head_elem` to `ucs_hlist_extract_head_elem`